src/libstd/error.rs

```diff
@@ -221,6 +222,24 @@ impl Error for ! {
     fn description(&self) -> &str { *self }
 }
 
+#[unstable(feature = "allocator_api",
+           reason = "the precise API and guarantees it provides may be tweaked.",
+           issue = "27700")]
+impl Error for allocator::AllocErr {
```

Comment:
I've added this as a tracking item on https://github.com/rust-lang/rust/issues/32838